netmaster/objApi/apiController.go

```diff
@@ -0,0 +1,639 @@
+func (ac *APIController) NetworkCreate(network *contivModel.Network) error {
+	log.Infof("Received NetworkCreate: %+v", network)
+
+	// Make sure tenant exists
+	if network.TenantName == "" {
+		return errors.New("Invalid tenant name")
+	}
+
+	tenant := contivModel.FindTenant(network.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
+	}
+
+	// Setup links
+	modeldb.AddLink(&network.Links.Tenant, tenant)
+	modeldb.AddLinkSet(&tenant.LinkSets.Networks, network)
+
+	// Save the tenant too since we added the links
+	err := tenant.Write()
+	if err != nil {
+		log.Errorf("Error updating tenant state(%+v). Err: %v", tenant, err)
+		return err
+	}
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return err
+	}
+
+	// Build networ config
+	networkCfg := intent.ConfigNetwork{
+		Name:       network.NetworkName,
+		PktTagType: network.Encap,
+		PktTag:     "",
+		SubnetCIDR: network.Subnet,
+		DefaultGw:  network.DefaultGw,
+	}
+
+	// Create the network
+	err = master.CreateNetwork(networkCfg, stateDriver, network.TenantName)
+	if err != nil {
+		log.Errorf("Error creating network {%+v}. Err: %v", network, err)
+		return err
+	}
+
+	return nil
+}
+
+// NetworkUpdate updates network
+func (ac *APIController) NetworkUpdate(network, params *contivModel.Network) error {
+	log.Infof("Received NetworkUpdate: %+v, params: %+v", network, params)
+	return errors.New("Cant change network parameters after its created")
+}
+
+// NetworkDelete deletes network
+func (ac *APIController) NetworkDelete(network *contivModel.Network) error {
+	log.Infof("Received NetworkDelete: %+v", network)
+
+	// Find the tenant
+	tenant := contivModel.FindTenant(network.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
+	}
+
+	// Remove link
+	modeldb.RemoveLinkSet(&tenant.LinkSets.Networks, network)
+
+	// Save the tenant too since we added the links
+	err := tenant.Write()
+	if err != nil {
+		return err
+	}
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return err
+	}
+
+	// Delete the network
+	err = master.DeleteNetworkID(stateDriver, network.NetworkName)
+	if err != nil {
+		log.Errorf("Error deleting network %s. Err: %v", network.NetworkName, err)
+	}
+
+	return nil
+}
+
+// PolicyCreate creates policy
+func (ac *APIController) PolicyCreate(policy *contivModel.Policy) error {
+	log.Infof("Received PolicyCreate: %+v", policy)
+	return nil
+}
+
+// PolicyUpdate updates policy
+func (ac *APIController) PolicyUpdate(policy, params *contivModel.Policy) error {
+	log.Infof("Received PolicyUpdate: %+v, params: %+v", policy, params)
+	return nil
+}
+
+// PolicyDelete deletes policy
+func (ac *APIController) PolicyDelete(policy *contivModel.Policy) error {
+	log.Infof("Received PolicyDelete: %+v", policy)
+	return nil
+}
+
+// ServiceCreate creates service
+func (ac *APIController) ServiceCreate(service *contivModel.Service) error {
```

Comment:
similar to comment for app handling, even service handling will move out of netmaster eventually, right?